### ISSUE TITLE
Missing street names in dataset for Czech Republic

### ISSUE THREAD

**Issue by JiriBalcar** [openaddresses/openaddresses#3429]
Hello, I found a problem with dataset for Czech Republic.

Street name in datasource is missing if street name in RUIAN (datasource) starts with number followed by dot. In this case is also housenumber always the number from the street name (for every house on that street).

For example this address:
https://www.google.cz/maps/place/28.+%C5%99%C3%ADjna+24,+Vnit%C5%99n%C3%AD+M%C4%9Bsto,+563+01+Lan%C5%A1kroun/@49.9126628,16.6131371,18z/data=!3m1!4b1!4m13!1m7!3m6!1s0x470df610ad4eb25d:0x73f57b32128c678c!2zMjguIMWZw61qbmEsIDU2MyAwMSBMYW7FoWtyb3Vu!3b1!8m2!3d49.9125382!4d16.6137194!3m4!1s0x470df617538f8981:0x77b4eb89fd7d22b4!8m2!3d49.9126628!4d16.6142341?hl=cs
```
28. října 24
563 01  Lanškroun

street_name = "28. října"
housenumber = "24"
city = "Lanškroun"
postalcode = "56301"
```
is parsed to csv dataset as:
```
LON,LAT,NUMBER,STREET,UNIT,CITY,DISTRICT,REGION,POSTCODE,ID,HASH
16.614226,49.912664,28,,,Lanškroun,,,56301,,defb0b3464239199
```
Another example:
https://www.google.cz/maps/place/1.+kv%C4%9Btna+683%2F86,+692+01+Mikulov/@48.811122,16.645507,18z/data=!3m1!4b1!4m13!1m7!3m6!1s0x0:0x0!2zNDjCsDQ4JzQwLjAiTiAxNsKwMzgnNDYuNSJF!3b1!8m2!3d48.811123!4d16.646237!3m4!1s0x4712cd343c007c6d:0xbb2b5e75568c2bd0!8m2!3d48.811122!4d16.6462392?hl=cs
```
1. května 683/86
692 01 Mikulov

street_name = "1. května"
housenumber = "683/86"
city = "Mikulov"
postalcode = "69201"
```

parsed as:
```
LON,LAT,NUMBER,STREET,UNIT,CITY,DISTRICT,REGION,POSTCODE,ID,HASH
16.646237,48.811123,1,,,Mikulov,,,69201,,c676e78fbe42fa57
```

I looked around I found that maybe the root cause of this problem is wrong ReqEx for CZ street
https://github.com/openaddresses/openaddresses/blob/85bad5e6559d0f50f36d2ea60cb9b6b7cb8732ea/sources/cz/countrywide.json#L42

But I could not find if there is any place where the address is checked with the regex.

**Comment by JiriBalcar** [comment]
I just found that this issue is not only if the street name start with number followed by dot, but if the street address contains this number.

Example:
https://www.google.cz/maps/place/t%C5%99.+5.+kv%C4%9Btna+7%2F101,+373+72+Li%C5%A1ov/@49.0161463,14.6023604,17z/data=!3m1!4b1!4m13!1m7!3m6!1s0x0:0x0!2zNDnCsDAwJzU4LjEiTiAxNMKwMzYnMTYuNCJF!3b1!8m2!3d49.016141!4d14.604553!3m4!1s0x47734bfaeb0ffb69:0x165548effb555b1e!8m2!3d49.0161463!4d14.6045544?hl=cs

```
třída 5. května 7
373 72  Lišov

street_name = "třída 5. května"
housenumber = "7"
city = "Lišov"
postalcode = "37372"
```

parsed as:
```
LON,LAT,NUMBER,STREET,UNIT,CITY,DISTRICT,REGION,POSTCODE,ID,HASH
14.604553,49.016141,5,třída,,Lišov,,,37372,,59352d355a354b52
```

**Comment by nvkelso** [comment]
@albarrentine Can you suggest some RegEx magic to fix this, please?

**Comment by JiriBalcar** [comment]
Is the place where you use the RegEx opensourced? I would like to take a look.

I did not found it anywhere. I tried cloning this repo to run on my machine but there is the scripts which generate only csv which I guess is somewhere processed with the RegEx. Also I did not found any code to translate coords from RUIAN datasource (S-JTSK / Krovak East North) to WGS.

**Comment by nvkelso** [comment]
Please see:

https://github.com/openaddresses/openaddresses/blob/324b2eb04cae3184119711e9c5fb150dd86f0c14/sources/cz/countrywide.json#L31-L57

And general background:

- https://github.com/openaddresses/openaddresses/blob/master/ATTRIBUTE_FUNCTIONS.md

**Comment by migurski** [comment]
You may also be wondering how to run the OpenAddresses code yourself?

- Each RegExp is applied here: https://github.com/openaddresses/machine/blob/1caafa8/openaddr/conform.py#L1006-L1016
- Documentation on running sources yourself can be found here: https://github.com/openaddresses/machine/blob/master/docs/install.md#running-a-source-locally

**Comment by albarrentine** [comment]
As long as we can rely on the comma delimiters in this specific data set (which appears to be true at the moment), it is possible with regex. Without commas it becomes generalized address parsing, at which point an NLP-based model like libpostal is needed.

Have something that works for the above examples, will submit a PR.

**Comment by trescube** [comment]
Don't forget the acceptance tests!  They love and care for you.

**Comment by albarrentine** [comment]
Fixed in #3431
